From c59f47f9ae1e6e03fdaa54737f31f58836b0ddbb Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sat, 19 Jul 2014 04:27:58 -0400
Subject: [pci] Remove unused pci_vpd_truncate()

Message-id: <20140719042758.19438.73023.stgit@gir.stowe>
Patchwork-id: 86012
O-Subject: [RHEL7.1 PATCH 50/81] PCI: Remove unused pci_vpd_truncate()
Bugzilla: 1110895
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110895
Upstream Status: 3984ca1c6e493d06fa7e8dc96ce2b23b86175ec4

commit 3984ca1c6e493d06fa7e8dc96ce2b23b86175ec4
Author: Stephen Hemminger <stephen@networkplumber.org>
Date:   Fri Jan 10 15:10:13 2014 -0700

    PCI: Remove unused pci_vpd_truncate()

    My philosophy is unused code is dead code.  And dead code is subject to bit
    rot and is a likely source of bugs.  Use it or lose it.

    This reverts db5679437a2b ("PCI: add interface to set visible size of
    VPD"), removing this interface:

        pci_vpd_truncate()

    [bhelgaas: split to separate patch, also remove prototype from pci.h]
    Signed-off-by: Stephen Hemminger <stephen@networkplumber.org>
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/access.c b/drivers/pci/access.c
index 4021b6c..ff25101 100644
--- a/drivers/pci/access.c
+++ b/drivers/pci/access.c
@@ -381,30 +381,6 @@ int pci_vpd_pci22_init(struct pci_dev *dev)
 }
 
 /**
- * pci_vpd_truncate - Set available Vital Product Data size
- * @dev: pci device struct
- * @size: available memory in bytes
- *
- * Adjust size of available VPD area.
- */
-int pci_vpd_truncate(struct pci_dev *dev, size_t size)
-{
- if (!dev->vpd)
-  return -EINVAL;
-
- /* limited by the access method */
- if (size > dev->vpd->len)
-  return -EINVAL;
-
- dev->vpd->len = size;
- if (dev->vpd->attr)
-  dev->vpd->attr->size = size;
-
- return 0;
-}
-EXPORT_SYMBOL(pci_vpd_truncate);
-
-/**
  * pci_cfg_access_lock - Lock PCI config reads/writes
  * @dev: pci device struct
  *
diff --git a/include/linux/pci.h b/include/linux/pci.h
index bd94582..b85a88a 100644
--- a/include/linux/pci.h
+++ b/include/linux/pci.h
@@ -1041,7 +1041,6 @@ unsigned int pci_rescan_bus(struct pci_bus *bus);
 /* Vital product data routines */
 ssize_t pci_read_vpd(struct pci_dev *dev, loff_t pos, size_t count, void *buf);
 ssize_t pci_write_vpd(struct pci_dev *dev, loff_t pos, size_t count, const void *buf);
-int pci_vpd_truncate(struct pci_dev *dev, size_t size);
 
 /* Helper functions for low-level code (drivers/pci/setup-[bus,res].c) */
 resource_size_t pcibios_retrieve_fw_addr(struct pci_dev *dev, int idx);
-- 
1.7.1